controllers/dockercluster_controller.go

```diff
@@ -38,88 +46,131 @@ type DockerClusterReconciler struct {
-		if err != nil {
-			log.Error(err, "Failed to get load balancer port")
-			return ctrl.Result{}, err
-		}
-		dockerCluster.Status.APIEndpoints = []infrastructurev1alpha2.APIEndpoint{
-			{
-				Host: ipv4,
-				Port: int(port),
-			},
-		}
+	// Handle deleted clusters
+	if !dockerCluster.DeletionTimestamp.IsZero() {
+		return reconcileDelete(lbsvc)
+	}
+
+	// Handle non-deleted clusters
+	return reconcileNormal(lbsvc)
+}
+
+func reconcileNormal(lbsvc *docker.LoadBalancer) (reconcile.Result, error) {
+	dockerCluster := lbsvc.DockerCluster
+
+	// If the DockerCluster doesn't have finalizer, add it.
+	if !util.Contains(dockerCluster.Finalizers, infrav1.ClusterFinalizer) {
+		dockerCluster.Finalizers = append(dockerCluster.Finalizers, infrav1.ClusterFinalizer)
+	}
+
+	//Create the docker container hosting the load balancer
+	n, err := lbsvc.Create()
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to reconcile load balancer for DockerCluster %s/%s", dockerCluster.Namespace, dockerCluster.Name)
+	}
+
+	// Set APIEndpoints with the load balancer IP so the Cluster API Cluster Controller can pull them
+	lbip4, _, err := n.IP()
+	if err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "failed to get ip for the load balancer for DockerCluster %s/%s", dockerCluster.Namespace, dockerCluster.Name)
+	}
+
+	dockerCluster.Status.APIEndpoints = []infrav1.APIEndpoint{
+		{
+			Host: lbip4,
+			Port: loadbalancer.ControlPlanePort,
+		},
 	}
 
-	log.Info("Reconcile network for cluster successful", "APIEndPoint", dockerCluster.Status.APIEndpoints[0])
+	// Mark the dockerCluster ready
 	dockerCluster.Status.Ready = true
-	return ctrl.Result{}, nil
+
+	return reconcile.Result{}, nil
+}
+
+func reconcileDelete(lbsvc *docker.LoadBalancer) (reconcile.Result, error) {
+	dockerCluster := lbsvc.DockerCluster
+
+	// Delete the docker container hosting the load balancer
+	if err := lbsvc.Delete(); err != nil {
+		return reconcile.Result{}, errors.Wrapf(err, "error deleting load balancer for DockerCluster %s/%s", dockerCluster.Namespace, dockerCluster.Name)
+	}
+
+	// Cluster is deleted so remove the finalizer.
+	dockerCluster.Finalizers = util.Filter(dockerCluster.Finalizers, infrav1.ClusterFinalizer)
+
+	return reconcile.Result{}, nil
 }
 
 // SetupWithManager will add watches for this controller
 func (r *DockerClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&infrastructurev1alpha2.DockerCluster{}).
+		For(&infrav1.DockerCluster{}).
 		Complete(r)
 }
 
-func (r *DockerClusterReconciler) patchCluster(ctx context.Context,
-	dockerCluster *infrastructurev1alpha2.DockerCluster, patchConfig client.Patch) error {
-	if err := r.Status().Patch(ctx, dockerCluster, patchConfig); err != nil {
-		return err
+func (r *DockerClusterReconciler) patchDockerCluster(ctx context.Context, dockerCluster *infrav1.DockerCluster, dockerClusterPatch client.Patch) error {
+	// Patch Cluster object.
```

Comment:
we don't need this comment or the comment on line 170. Or we want this to be consistent with the code and say "patch docker cluster object", but I think that's pretty evident from the code.